gammapy/irf/tests/test_background.py

```diff
@@ -220,6 +220,39 @@ def test_background_3d_read_gadf():
     assert data.unit == "s-1 MeV-1 sr-1"
 
 
+def test_bkg_3d_wrong_units():
+    energy = [0.1, 10, 1000] * u.TeV
+    energy_axis = MapAxis.from_energy_edges(energy)
+
+    fov_lon = [0, 1, 2, 3] * u.deg
+    fov_lon_axis = MapAxis.from_edges(fov_lon, name="fov_lon")
+
+    fov_lat = [0, 1, 2, 3] * u.deg
+    fov_lat_axis = MapAxis.from_edges(fov_lat, name="fov_lat")
+
+    wrong_unit = u.cm**2 * u.s
+    data = np.ones((2, 3, 3)) * wrong_unit
+    with pytest.raises(ValueError) as error:
+        Background3D(axes=[energy_axis, fov_lon_axis, fov_lat_axis],
+                 data=data)
+    assert error.match(f"Error: (.*) is not an allowed unit. (.*) requires (.*) data quantities.")
```

Comment:
This probably does not need the f-string. Apparently this breaks the py flakes CI.